src/runner.rs

```diff
@@ -909,66 +907,44 @@ impl Log for RunnerLogger {
             return;
         }
 
-        let binding = record.args().to_string();
-        let msg_parts: Vec<&str> = binding.split_whitespace().collect();
-
-        if msg_parts.len() >= 4 {
-            println!(
-                "{} {} {} [{}] {}",
-                msg_parts[0],
-                msg_parts[1],
-                msg_parts[2],
-                record.target(),
-                msg_parts[3..].join(" ")
-            );
+        if let Some(log_file) = &self.log_file {
+            let mut log_file: MutexGuard<'_, File> = log_file.lock().unwrap();
```

Comment:
nit: you can get rid of the explicit `MutexGuard` type here and let type inference figure it out.